packages/client/src/WalletAPIClient.ts

```diff
@@ -33,10 +34,12 @@ const requestHandlers = {
 /**
  * WalletAPI Client which rely on WindowMessage communication
  */
-export class WalletAPIClient extends RpcNode<
-  typeof requestHandlers,
-  WalletHandlers
-> {
+export class WalletAPIClient<
+  M extends CustomModule = CustomModule,
+  CustomGetter extends (client: WalletAPIClient<M>) => M | Record<string, M> = (
+    client: WalletAPIClient<M>,
+  ) => M | Record<string, M>,
+> extends RpcNode<typeof requestHandlers, WalletHandlers> {
```

Comment:
What do you think of this decomposition for the sake of readability?

```
type CustomGetterType<M extends CustomModule> = (client: WalletAPIClient<M>) => M | Record<string, M>;

export class WalletAPIClient<
  M extends CustomModule = CustomModule,
  CustomGetter extends CustomGetterType<M> = CustomGetterType<M>
> extends RpcNode<typeof requestHandlers, WalletHandlers> {
```

Reply:
I need to test again but it was breaking inferance

Reply:
Ah damn :( Can we pair to try again?